src/Illuminate/Conditionable/HigherOrderWhenProxy.php

```diff
@@ -18,17 +18,55 @@ class HigherOrderWhenProxy
      */
     protected $condition;
 
+    /**
+     * Tracks whether the proxy has a condition.
+     *
+     * @var bool
+     */
+    protected $hasCondition = false;
+
+    /**
+     * Determine whether the condition should be negated.
+     *
+     * @var bool
+     */
+    protected $negateCondition;
+
     /**
      * Create a new proxy instance.
      *
      * @param  mixed  $target
-     * @param  bool  $condition
      * @return void
      */
-    public function __construct($target, $condition)
```

Comment:
Removing this is a breaking change.

Reply:
You are right that it is a breaking change, though isn't it considered part of the internal API?